core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -878,14 +965,19 @@ class ReplicaManager(val config: KafkaConfig,
         else
           Set.empty[Partition]
 
+        leaderAndISRRequest.partitionStates.asScala.keys.foreach( topicPartition =>
+          // Remove the partition from cache if the local replica creation has failed due to KafkaStorageException
```

Comment:
I see. I replaced the comment with this: `Mark partition as offline in the cache if the local replica creation has failed due to KafkaStorageException`